packages/dts-test/ref.test-d.ts

```diff
@@ -201,11 +201,23 @@ if (refStatus.value === 'initial') {
   expectType<IsAny<typeof a>>(false)
 }
 
-describe('shallowRef with generic', <T>() => {
-  const r = ref({}) as MaybeRef<T>
```

Comment:
This change does break the `MaybeRef<T>` case, but the correctness improvement is worth the tradeoff IMO.